src/pages/Edit/HashtagBox/index.jsx

```diff
@@ -0,0 +1,69 @@
+import { useState } from 'react';
+
+import * as S from './index.styles';
+
+import { CLIENT_MESSAGE } from '@/constants/message';
+import useCheckBlack from '@/hooks/useCheckBlank';
+
+const HashtagBox = ({ hashtagList, setHashtagList, showSnackbar }) => {
+  const [newHashtag, setNewHashtag] = useState('');
+  const { isBlank, isNull } = useCheckBlack();
+  const onChangeHashtag = e => {
+    setNewHashtag(e.target.value);
+  };
+  const addHashtag = ({ key }) => {
+    if (key === 'Enter') {
+      if (isBlank(newHashtag) || isNull(newHashtag)) {
+        showSnackbar(CLIENT_MESSAGE.ERROR.EMPTY_HASHTAG);
+        setNewHashtag('');
+        return;
+      }
+      if (hashtagList.length === 5) {
+        showSnackbar(CLIENT_MESSAGE.ERROR.FULL_HASHTAG_LIST);
+        setNewHashtag('');
+        return;
+      }
+      if (hashtagList.includes(newHashtag)) {
+        showSnackbar(CLIENT_MESSAGE.ERROR.REPUTATION_HASHTAG);
+        setNewHashtag('');
+        return;
+      }
+      setHashtagList(prev => [...prev, newHashtag]);
+      setNewHashtag('');
+    }
```

Comment:
if 문 안의 내용이 길어지면 결국 한 뎁스 더 있는 구문이 길어져서 보기가 조금 힘들어요. 그래서 클린코드 관점에서도 예외처리를 먼저 하라는 게 있는 것 같아요,,
이 부분에서는 key가 'Enter'가 아닐 땐 아무것도 하지 않으니 그럴 때를 먼저 예외처리 해주면 훨씬 나아져요.

```suggestion
    if (key !== 'Enter') return;

    if (isBlank(newHashtag) || isNull(newHashtag)) {
      showSnackbar(CLIENT_MESSAGE.ERROR.EMPTY_HASHTAG);
      setNewHashtag('');
      
      return;
    }
    if (hashtagList.length === 5) {
    showSnackbar(CLIENT_MESSAGE.ERROR.FULL_HASHTAG_LIST);
      setNewHashtag('');
      
      return;
    }
    if (hashtagList.includes(newHashtag)) {
    showSnackbar(CLIENT_MESSAGE.ERROR.REPUTATION_HASHTAG);
      setNewHashtag('');
      
      return;
    }
    setHashtagList(prev => [...prev, newHashtag]);
    setNewHashtag('');
```

Reply:
그리고 이건 UX와 관련된 것이긴 한데요. 사용자가 잘못 입력해서 무언가 제출하는 트리거를 했을 때, (저희의 경우는 엔터겠죠?) 내가 뭘 입력해서 잘못한건지 기억나지도 않는데 개발자가 잘못 입력한 경우는 무조건 입력한 값을 초기화하도록 만들어서 매우 불편하다는 게 요즘 종종 등장하는 피드백입니다. 그래서 요즘은 잘못 입력해도 input을 잘 비워주지 않습니다. 여기도 마찬가지로 비워주지 않는 방식으로 가면 좋을 것 같아요. 비워주는 건 제대로 입력해서 추가됐을 때에 한해서!

Reply:
추가적으로 에러 발견했습니다. 해시태그가 한글일 때는 추가하면 `해시태그 값을 입력해주세요` 에러가 발생하네요.

Reply:
key가 enter가 아닐 때 예외처리 먼저,
잘못된 해시태그를 입력할 땐 input 그대로 두기
한글 입력 에러 해결
확인했습니다!